src/main/java/ivanov/springbootintro/mapper/BookMapper.java

```diff
@@ -32,12 +33,16 @@ default void setCategoryIds(@MappingTarget BookDto bookDto, Book book) {
 
     @AfterMapping
     default void setCategories(@MappingTarget Book book, CreateBookRequestDto requestDto) {
-        book.setCategories(requestDto.categoryIds().stream()
-                .map(categoryId -> {
-                    Category category = new Category();
-                    category.setId(categoryId);
-                    return category;
-                })
-                .collect(Collectors.toSet()));
+        if (requestDto.categoryIds() != null) {

```

Comment:
You can use the Objects.nonNull()
```suggestion
        if (Objects.nonNull(requestDto.categoryIds())) {
```